x-pack/plugins/cases/public/components/create/assignees.tsx

```diff
@@ -0,0 +1,216 @@
+import { useAvailableCasesOwners } from '../app/use_available_owners';
+import { getAllPermissionsExceptFrom } from '../../utils/permissions';
+
+interface Props {
+  isLoading: boolean;
+}
+
+interface FieldProps {
+  field: FieldHook;
+  options: EuiComboBoxOptionOption[];
+  isLoading: boolean;
+  isDisabled: boolean;
+  currentUserProfile: UserProfile;
+  selectedOptions: EuiComboBoxOptionOption[];
+  setSelectedOptions: React.Dispatch<React.SetStateAction<EuiComboBoxOptionOption[]>>;
+  onSearchComboChange: (value: string) => void;
+}
+
+const getConfig = (): FieldConfig => ({
+  label: i18n.ASSIGNEES,
+  defaultValue: [],
+});
+
+const userProfileToComboBoxOption = (userProfile: UserProfileWithAvatar) => ({
+  label: getUserDisplayName(userProfile.user),
+  value: userProfile.uid,
+  user: userProfile.user,
+  data: userProfile.data,
+});
+
+const comboBoxOptionToAssignee = (option: EuiComboBoxOptionOption) => ({ uid: option.value });
+
+const AssigneesFieldComponent: React.FC<FieldProps> = React.memo(
+  ({
+    field,
+    isLoading,
+    isDisabled,
+    options,
+    currentUserProfile,
+    selectedOptions,
+    setSelectedOptions,
+    onSearchComboChange,
+  }) => {
+    const { setValue } = field;
+
+    const onComboChange = useCallback(
+      (currentOptions: EuiComboBoxOptionOption[]) => {
+        setSelectedOptions(currentOptions);
+        setValue(currentOptions.map((option) => comboBoxOptionToAssignee(option)));
+      },
+      [setSelectedOptions, setValue]
+    );
+
+    const onSelfAssign = useCallback(() => {
+      if (!currentUserProfile) {
+        return;
+      }
+
+      setSelectedOptions((prev) => [
+        ...(prev ?? []),
+        userProfileToComboBoxOption(currentUserProfile),
+      ]);
+
+      setValue([
+        ...(selectedOptions?.map((option) => comboBoxOptionToAssignee(option)) ?? []),
+        { uid: currentUserProfile.uid },
+      ]);
+    }, [currentUserProfile, selectedOptions, setSelectedOptions, setValue]);
+
+    const renderOption = useCallback(
+      (option: EuiComboBoxOptionOption, searchValue: string, contentClassName: string) => {
+        const { user, data, value } = option as EuiComboBoxOptionOption<string> &
+          UserProfileWithAvatar;
+
+        return (
+          <EuiSelectableListItem
+            key={value}
+            prepend={<UserAvatar user={user} avatar={data.avatar} size="s" />}
+            className={contentClassName}
+            append={<EuiTextColor color="subdued">{user.email}</EuiTextColor>}
+          >
+            {getUserDisplayName(user)}
+          </EuiSelectableListItem>
+        );
+      },
+      []
+    );
+
+    return (
+      <EuiFormRow
+        id="createCaseAssignees"
+        fullWidth
+        label={i18n.ASSIGNEES}
+        labelAppend={OptionalFieldLabel}
+        helpText={
+          <EuiLink data-test-subj="create-case-assign-yourself-link" onClick={onSelfAssign}>
+            {i18n.ASSIGN_YOURSELF}
```

Comment:
<img width="430" alt="Screen Shot 2022-09-08 at 3 22 36 PM" src="https://user-images.githubusercontent.com/21210601/189208735-a96be7d5-a58e-4257-96c2-b9e01eb3f517.png">

`Assign Yourself` link is available even when creating a Case as a User that does not have a User Profile, such as when using the Anonymous Authentication Provider

This link is correctly hidden on the Edit Case view and should probably be hidden here as well.

Reply:
Oops, thanks for catching that!

Reply:
https://github.com/elastic/kibana/pull/140208/commits/63ca0a7837fd99edbc127453f67fddc1176fff4a